idgenerator/aws/aws_xray_idgenerator.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws // import "go.opentelemetry.io/contrib/idgenerator/aws"
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type spanID [8]byte
+type traceID [16]byte
+
+type idGenerator interface {
+	NewTraceID() traceID
+	NewSpanID() spanID
+}
+
+type xRayIDGenerator struct {
+	sync.Mutex
+	randSource *rand.Rand
+}
+
+func awsXRayIDGenerator() idGenerator {
+	gen := &xRayIDGenerator{}
+	var rngSeed int64
+	err := binary.Read(crand.Reader, binary.LittleEndian, &rngSeed)
+	if err != nil {
+		panic(err)
```

Comment:
This cannot panic, instead it should return an error an a non-usable generator.

Reply:
Hi @Aneurysm9, for this function I took the same structure as the `defIDGenerator()` from the [trace.go file](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/trace/trace.go#L25-L31) in the main Go repo. Do you know what the convention is in this situation? Since I think in the [trace.go implementation](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/trace/trace.go#L25-L31) they just ignore the error.